app/src/main/java/com/randomappsinc/studentpicker/importdata/ImportFromFileActivity.java

```diff
@@ -58,55 +62,26 @@ public void onConfigurationChanged(@NonNull Configuration newConfig) {
         bannerAdManager.onOrientationChanged();
     }
 
-    private void extractNameListInfoFromTextFile() {
-        Handler handler = new Handler();
-        handler.post(() -> {
-            Uri fileUri = Uri.parse(getIntent().getStringExtra(Constants.FILE_URI_KEY));
-            Cursor cursor = getContentResolver().query(
-                    fileUri,
-                    null,
-                    null,
-                    null,
-                    null,
-                    null);
-
-            String listNameText = "";
-            if (cursor != null && cursor.moveToFirst()) {
-                String displayName = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME));
-                listNameText = displayName.replace(".txt", "");
-                cursor.close();
-            }
-
-            StringBuilder namesText = new StringBuilder();
-            InputStream inputStream = null;
-            BufferedReader reader = null;
-            try {
-                inputStream = getContentResolver().openInputStream(fileUri);
-                if (inputStream == null) {
-                    throw new IOException("Unable to find .txt file!");
-                }
-                reader = new BufferedReader(new InputStreamReader(inputStream));
-                String line;
-                while ((line = reader.readLine()) != null) {
-                    if (namesText.length() > 0) {
-                        namesText.append("\n");
-                    }
-                    namesText.append(line);
-                }
-                loadUI(listNameText, namesText.toString());
-            } catch (IOException exception) {
-                runOnUiThread(() -> UIUtils.showLongToast(
-                        R.string.load_file_fail, getApplicationContext()));
-                try {
-                    if (inputStream != null) {
-                        inputStream.close();
-                    }
-                    if (reader != null) {
-                        reader.close();
-                    }
-                } catch (IOException ignored) {}
-            }
-        });
+    private void extractNamesList(int fileType) {
+        Uri fileUri = Uri.parse(getIntent().getStringExtra(Constants.FILE_URI_KEY));
+        switch (fileType) {
+            case FileImportType.TEXT:
+                fileParsingManager.extractNameListFromText(this, fileUri);
+                break;
+            case FileImportType.CSV:
+                fileParsingManager.extractNameListFromCsv(this, fileUri);
+                break;
+        }
+    }
+
+    @Override
+    public void onFileParsingFailure(int fileParsingError) {
```

Comment:
Also, I think we should UI thread this since Toast is a UI change and this can be called on the background thread right?
```suggestion
    public void onFileParsingFailure(@StringRes int fileParsingError) {
```